Added 74779 8-bit Up/Down Counter

aaraujo666 opened the pull request:
New chip (Hopefully I didn't screw this one up :) )

aaraujo666 commented:
The 7428 is probably the same as the 7402 :) I only created it because I
use it in one of my circuits and saw that you didn't have it. Didn't
realize it was the same.

The DIL case inside the 74779... Maybe you can help... I had originally
created it without the DIL case, but then the two outputs (Q and ~Q)
wouldn't show up, when I used the Detail A inside the 74779. The only way I
found to "expose" the two outputs, was to make it a DIL and assign pin
numbers to the inputs and outputs. Is there another way to make the outputs
"visible" to the outside world without doing that?

As for the test case for the 74779. I have not written any test cases in
your tool yet. Is there documentation on how to do that? I will gladly
create them once I figure out how. As far as whether the chip works, I'm
pretty sure it does, since I am using it, as-is, in one of my designs.

On Tue, Feb 20, 2018 at 2:21 AM, Helmut Neemann <notifications@github.com>
wrote:

> New chip (Hopefully I didn't screw this one up :) )
>
> 74779 is far to complex to find a bug just by looking at it! ;-)
>
> But there are some questions:
>
>    - What is the difference between 7402 and 7428?
>    - Why did you put the inner circuit of 74779 in a DIL case? This makes
>    the circuit harder to read.
>
> The 74779 is a complex chip.There should be some test cases to make sure
> the chip works correctly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hneemann/Digital/pull/113#issuecomment-366887778>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAoF1tlA1JLP9WEeBHAihX8B4dWV9jzlks5tWnKRgaJpZM4SLVaz>
> .
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

hneemann commented:
> The 7428 is probably the same as the 7402 :) I only created it because I use it in one of my circuits and saw that you didn't have it. Didn't realize it was the same.

In the meantime i figured out the difference: Logical they are the same, but the 7428 is able to drive higher currents.

> The DIL case inside the 74779... Maybe you can help... I had originally created it without the DIL case, but then the two outputs (Q and ~Q) wouldn't show up, when I used the Detail A inside the 74779. The only way I found to "expose" the two outputs, was to make it a DIL and assign pin numbers to the inputs and outputs. Is there another way to make the outputs "visible" to the outside world without doing that?

I disabled the "DIL" check box, and the two outputs showed up immediately. Can you provide a test circuit which is not working and maybe a screenshot?

> As for the test case for the 74779. I have not written any test cases in your tool yet. Is there documentation on how to do that? I will gladly create them once I figure out how. As far as whether the chip works, I'm pretty sure it does, since I am using it, as-is, in one of my designs.

You can add a "Test" component to the circuit. (Components->misc->Test) The test data editor provides a help button which explains the usage. And most of the examples are also containing tests.

But there was a feature missing: So far it was not possible to test circuits which use bidirectional inputs. I have fixed this in the latest [pre release](https://infdigital.dhbw-mosbach.de/) version. You have to use the latest pre release if you want to create such tests. An explanation of how to do that is included in the help text.

Here is an [example](https://github.com/hneemann/Digital/blob/master/src/test/resources/dig/test/inAsOutTest.dig) (which only works with the latest [pre release](https://infdigital.dhbw-mosbach.de/) version) which shows the test of a circuit with a bidirectional input pin.

aaraujo666 commented:
Ok...

Working on a 74LS181 (4-bit ALU). Creating the tests...

So... for S0-3 = 0000, ALU returns whatever is in the A inputs. So if A0-3
1010, F0-3 wil be 1010. Test is as follows:

S3 S2 S1 S0 M Cn ~A3 ~A2 ~A1 ~A0 ~B3 ~B2 ~B1 ~B0 ~F3 ~F2 ~F1 ~F0

repeat(16) 0  0  0  0  0  1   bits(4,n)   X   X   X   X   bits(4,n)

The next operation (A or B) (S0-3 = 0001) I was wondering if it is possible
to use boolean operations in the test, similar to the example in the help:

loop(a,16)
  loop(b,16)
             0 0 0 1 0 1 bits(4,a) bits(4,b) bits(4,a|b)
  end loop
end loop

and (A and B) would be:

loop(a,16)
  loop(b,16)
             0 0 0 1 0 1 bits(4,a) bits(4,b) bits(4,a&b)
  end loop
end loop

Note the "|" (or) operator and the "&" (and) operator that I would like to
use...

Similarly "~" for NOT, would be nice...

Thanks

Tony


On Tue, Feb 20, 2018 at 10:58 AM, Helmut Neemann <notifications@github.com>
wrote:

> The 7428 is probably the same as the 7402 :) I only created it because I
> use it in one of my circuits and saw that you didn't have it. Didn't
> realize it was the same.
>
> In the meantime i figured out the difference: Logical they are the same,
> but the 7428 is able to drive higher currents.
>
> The DIL case inside the 74779... Maybe you can help... I had originally
> created it without the DIL case, but then the two outputs (Q and ~Q)
> wouldn't show up, when I used the Detail A inside the 74779. The only way I
> found to "expose" the two outputs, was to make it a DIL and assign pin
> numbers to the inputs and outputs. Is there another way to make the outputs
> "visible" to the outside world without doing that?
>
> I disabled the "DIL" check box, and the two outputs showed up immediately.
> Can you provide a test circuit which is not working and maybe a screenshot?
>
> As for the test case for the 74779. I have not written any test cases in
> your tool yet. Is there documentation on how to do that? I will gladly
> create them once I figure out how. As far as whether the chip works, I'm
> pretty sure it does, since I am using it, as-is, in one of my designs.
>
> You can add a "Test" component to the circuit. (Components->misc->Test)
> The test data editor provides a help button which explains the usage. And
> most of the examples are also containing tests.
>
> But there was a feature missing: So far it was not possible to test
> circuits which use bidirectional inputs. I have fixed this in the latest pre
> release <https://infdigital.dhbw-mosbach.de/> version. You have to use
> the latest pre release if you want to create such tests. An explanation of
> how to do that is included in the help text.
>
> Here is an example
> <https://github.com/hneemann/Digital/blob/master/src/test/resources/dig/test/inAsOutTest.dig>
> (which only works with the latest pre release
> <https://infdigital.dhbw-mosbach.de/> version) which shows the test of a
> circuit with a bidirectional input pin.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hneemann/Digital/pull/113#issuecomment-367023981>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAoF1iuJrK0iiClPtIkrpFazG35DrnNkks5tWuuIgaJpZM4SLVaz>
> .
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

aaraujo666 commented:
Never mind... It already works that way :)

On Tue, Feb 20, 2018 at 11:50 PM, Anthony Araujo <anthonyaraujo@gmail.com>
wrote:

> Ok...
>
> Working on a 74LS181 (4-bit ALU). Creating the tests...
>
> So... for S0-3 = 0000, ALU returns whatever is in the A inputs. So if A0-3
> 1010, F0-3 wil be 1010. Test is as follows:
>
> S3 S2 S1 S0 M Cn ~A3 ~A2 ~A1 ~A0 ~B3 ~B2 ~B1 ~B0 ~F3 ~F2 ~F1 ~F0
>
> repeat(16) 0  0  0  0  0  1   bits(4,n)   X   X   X   X   bits(4,n)
>
> The next operation (A or B) (S0-3 = 0001) I was wondering if it is
> possible to use boolean operations in the test, similar to the example in
> the help:
>
> loop(a,16)
>   loop(b,16)
>              0 0 0 1 0 1 bits(4,a) bits(4,b) bits(4,a|b)
>   end loop
> end loop
>
> and (A and B) would be:
>
> loop(a,16)
>   loop(b,16)
>              0 0 0 1 0 1 bits(4,a) bits(4,b) bits(4,a&b)
>   end loop
> end loop
>
> Note the "|" (or) operator and the "&" (and) operator that I would like to
> use...
>
> Similarly "~" for NOT, would be nice...
>
> Thanks
>
> Tony
>
>
> On Tue, Feb 20, 2018 at 10:58 AM, Helmut Neemann <notifications@github.com
> > wrote:
>
>> The 7428 is probably the same as the 7402 :) I only created it because I
>> use it in one of my circuits and saw that you didn't have it. Didn't
>> realize it was the same.
>>
>> In the meantime i figured out the difference: Logical they are the same,
>> but the 7428 is able to drive higher currents.
>>
>> The DIL case inside the 74779... Maybe you can help... I had originally
>> created it without the DIL case, but then the two outputs (Q and ~Q)
>> wouldn't show up, when I used the Detail A inside the 74779. The only way I
>> found to "expose" the two outputs, was to make it a DIL and assign pin
>> numbers to the inputs and outputs. Is there another way to make the outputs
>> "visible" to the outside world without doing that?
>>
>> I disabled the "DIL" check box, and the two outputs showed up
>> immediately. Can you provide a test circuit which is not working and maybe
>> a screenshot?
>>
>> As for the test case for the 74779. I have not written any test cases in
>> your tool yet. Is there documentation on how to do that? I will gladly
>> create them once I figure out how. As far as whether the chip works, I'm
>> pretty sure it does, since I am using it, as-is, in one of my designs.
>>
>> You can add a "Test" component to the circuit. (Components->misc->Test)
>> The test data editor provides a help button which explains the usage. And
>> most of the examples are also containing tests.
>>
>> But there was a feature missing: So far it was not possible to test
>> circuits which use bidirectional inputs. I have fixed this in the latest pre
>> release <https://infdigital.dhbw-mosbach.de/> version. You have to use
>> the latest pre release if you want to create such tests. An explanation of
>> how to do that is included in the help text.
>>
>> Here is an example
>> <https://github.com/hneemann/Digital/blob/master/src/test/resources/dig/test/inAsOutTest.dig>
>> (which only works with the latest pre release
>> <https://infdigital.dhbw-mosbach.de/> version) which shows the test of a
>> circuit with a bidirectional input pin.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/hneemann/Digital/pull/113#issuecomment-367023981>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAoF1iuJrK0iiClPtIkrpFazG35DrnNkks5tWuuIgaJpZM4SLVaz>
>> .
>>
>
>
>
> --
> Anthony de Araujo
> anthonyaraujo@gmail.com
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

aaraujo666 commented:
Any idea why my 74273 is not available in my library?

[image: Inline image 1]

On Thu, Feb 22, 2018 at 8:45 AM, Helmut Neemann <notifications@github.com>
wrote:

> Closed #113 <https://github.com/hneemann/Digital/pull/113> via 76f8a16
> <https://github.com/hneemann/Digital/commit/76f8a16bab5f10b22fd1595804480d6cc00f8fbb>
> .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hneemann/Digital/pull/113#event-1486962806>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAoF1hcMM3fk_gh04t-uEq4iEDpLxWbOks5tXW9ogaJpZM4SLVaz>
> .
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

hneemann commented:
I have moved the 74273 to the folder "flipflops". If in your local copy the 74272 is still in the folder "memory" and also in the folder "flipflops" then there are two files with the same name "74273.dig".
In this case the name is not unique and it is not possible to select on of the files.
Because Digital uses only the file name and not the file path to identify a file, the names must be unique.
Maybe this is the reason.

aaraujo666 commented:
I did have two copies as you suspected... But I deleted my version in the
memory folder, quit Digital, and reloaded it and it still shows up with the
X...

Is what it read from the library "cached" anywhere? Maybe it thinks it's
still there? Or does it reload the entire library upon startup?

On Thu, Feb 22, 2018 at 10:14 AM, Helmut Neemann <notifications@github.com>
wrote:

> I have moved the 74273 to the folder "flipflops". If in your local copy
> the 74272 is still in the folder "memory" and also in the folder
> "flipflops" then there are two files with the same name "74273.dig".
> In this case the name is not unique and it is not possible to select on of
> the files.
> Because Digital uses only the file name and not the file path to identify
> a file, the names must be unique.
> Maybe this is the reason.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hneemann/Digital/pull/113#issuecomment-367712000>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAoF1hvB4C4iYWck8hK-q9FyuttKP7bXks5tXYQ9gaJpZM4SLVaz>
> .
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

aaraujo666 commented:
Completely deleted my Digital folder...

Downloaded latest zip from Github... Unzipped it to a new folder...

Loaded Digital, 74273 is still X and, for some reason, the 7489 which is
alone in the memory folder now, is also X

On Thu, Feb 22, 2018 at 10:14 AM, Helmut Neemann <notifications@github.com>
wrote:

> I have moved the 74273 to the folder "flipflops". If in your local copy
> the 74272 is still in the folder "memory" and also in the folder
> "flipflops" then there are two files with the same name "74273.dig".
> In this case the name is not unique and it is not possible to select on of
> the files.
> Because Digital uses only the file name and not the file path to identify
> a file, the names must be unique.
> Maybe this is the reason.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hneemann/Digital/pull/113#issuecomment-367712000>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAoF1hvB4C4iYWck8hK-q9FyuttKP7bXks5tXYQ9gaJpZM4SLVaz>
> .
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

aaraujo666 commented:
I think I figured it out...

I had a 74273.dig file open in the editor... whenever I would run, it would
reload THAT file since it was the last one open. I think it was the open
file conflicting with the library file (still doesn't explain why 7489 got
Xed).



On Thu, Feb 22, 2018 at 10:28 AM, Anthony Araujo <anthonyaraujo@gmail.com>
wrote:

> Completely deleted my Digital folder...
>
> Downloaded latest zip from Github... Unzipped it to a new folder...
>
> Loaded Digital, 74273 is still X and, for some reason, the 7489 which is
> alone in the memory folder now, is also X
>
> On Thu, Feb 22, 2018 at 10:14 AM, Helmut Neemann <notifications@github.com
> > wrote:
>
>> I have moved the 74273 to the folder "flipflops". If in your local copy
>> the 74272 is still in the folder "memory" and also in the folder
>> "flipflops" then there are two files with the same name "74273.dig".
>> In this case the name is not unique and it is not possible to select on
>> of the files.
>> Because Digital uses only the file name and not the file path to identify
>> a file, the names must be unique.
>> Maybe this is the reason.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/hneemann/Digital/pull/113#issuecomment-367712000>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAoF1hvB4C4iYWck8hK-q9FyuttKP7bXks5tXYQ9gaJpZM4SLVaz>
>> .
>>
>
>
>
> --
> Anthony de Araujo
> anthonyaraujo@gmail.com
>



-- 
Anthony de Araujo
anthonyaraujo@gmail.com

hneemann commented:
Digital does not store any information about library files.
Every time you start Digital, the folder is scanned again.
But Digital scans two folders: the library folder and the folder of the current file.
If you open a file from the library, both folders overlap. Digital should detect that and behave correct (no X). But maybe there is a bug somewhere.
Try to open a file which is not part of the library. Then close Digital and reopen it. After that the X should disappear.

hneemann commented:
I have played around on my machine and it's possible to open a library file without getting the red X (at least on Linux).
Are you able to reproduce the bug and post a description so that I can fix it?

aaraujo666 commented:
On Thu, Feb 22, 2018 at 11:19 AM Helmut Neemann <notifications@github.com>
wrote:

> I have played around on my machine and it's possible to open a library
> file without getting the red X (at least on Linux).
> Are you able to reproduce the bug and post a description so that I can fix
> it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hneemann/Digital/pull/113#issuecomment-367734326>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAoF1qarhcZCoFldKLeKlEF7GEzEvsIYks5tXZOngaJpZM4SLVaz>
> .
>
I’ll see if I can get a reproducible list of steps to help you locate the
issue (if there is one)
-- 
Anthony de Araujo
anthonyaraujo@gmail.com
